src/pytorch_lightning/utilities/device_parser.py

```diff
@@ -340,7 +341,7 @@ def num_cuda_devices() -> int:
     Unlike :func:`torch.cuda.device_count`, this function will do its best not to create a CUDA context for fork
     support, if the platform allows it.
     """
-    if "fork" not in torch.multiprocessing.get_all_start_methods():
+    if "fork" not in torch.multiprocessing.get_all_start_methods() or _is_forking_disabled():
```

Comment:
shouldn't this `if "fork" not in torch.multiprocessing.get_all_start_methods()` not be part of `_is_forking_disabled`? Doesn't matter if it's disabled due to the env variable or because it's not supported by the OS

Reply:
Maybe it could, but we also have this kind of check in the accelerator connector 

```py
        if strategy_flag in _DDP_FORK_ALIASES and "fork" not in torch.multiprocessing.get_all_start_methods():
            raise ValueError(
                f"You selected `Trainer(strategy='{strategy_flag}')` but process forking is not supported on this"
                f" platform. We recommed `Trainer(strategy='ddp_spawn')` instead."
            )
```

for a specific user message about platform support. So in other words, at the moment "fork not supported by platform" and "fork disabled" are two different error messages.